Source/AviationWeather.Test/Source/metar_parser_tests.cpp

```diff
@@ -55,488 +55,418 @@ TEST_CLASS(MetarParserTests)
-        std::string m1("METAR ");
-        metar::parsers::parse_report_type(metar, m1);
-        Assert::AreEqual(metar::report_type::metar, metar.type);
-    }
+        parsed = true;
+        Assert::AreEqual(metar_report_type::metar, type);
+    });
+    Assert::IsTrue(parsed);
 
     // SPECI
+    parsed = false;
+    parse_metar_report_type(std::string("SPECI "), [&](metar_report_type type)
     {
-        metar::metar_info metar("");
-        std::string m2("SPECI ");
-        metar::parsers::parse_report_type(metar, m2);
-        Assert::AreEqual(metar::report_type::special, metar.type);
-    }
+        parsed = true;
+        Assert::AreEqual(metar_report_type::special, type);
+    });
+    Assert::IsTrue(parsed);
 }
 
 //-----------------------------------------------------------------------------
 
 void MetarParserTests::METAR_Parser_StationIdentifier()
 {
+    // Empty
+    parse_station_identifier(std::string(""), [&](std::string const& identifier)
     {
-        metar::metar_info metar("");
-        Assert::AreEqual(std::string(""), metar.station_identifier);
-    }
+        Assert::Fail();
+    });
 
     // KSFO
+    bool parsed = false;
+    parse_station_identifier(std::string("KSFO "), [&](std::string const& identifier)
     {
-        metar::metar_info metar("");
-        std::string m1("KSFO ");
-        metar::parsers::parse_station_identifier(metar, m1);
-        Assert::AreEqual(std::string("KSFO"), metar.station_identifier);
-    }
+        parsed = true;
+        Assert::AreEqual(std::string("KSFO"), identifier);
+    });
+    Assert::IsTrue(parsed);
 
     // EBGE
+    parsed = false;
+    parse_station_identifier(std::string("EGBE "), [&](std::string const& identifier)
     {
-        metar::metar_info metar("");
-        std::string m2("EGBE ");
-        metar::parsers::parse_station_identifier(metar, m2);
-        Assert::AreEqual(std::string("EGBE"), metar.station_identifier);
-    }
+        parsed = true;
+        Assert::AreEqual(std::string("EGBE"), identifier);
+    });
+    Assert::IsTrue(parsed);
 }
 
 //-----------------------------------------------------------------------------
 
 void MetarParserTests::METAR_Parser_ObservationTime()
 {
+    // Empty
+    parse_time(std::string(""), [&](aw::time && time)
+    {
+        Assert::Fail();
+    });
+
     // 8th, 17:53
+    parse_time(std::string("081753Z "), [&](aw::time && time)
     {
-        metar::metar_info metar("");
-        std::string m1("081753Z ");
-        metar::parsers::parse_observation_time(metar, m1);
-        Assert::AreEqual(uint8_t(8), metar.report_time.day_of_month);
-        Assert::AreEqual(uint8_t(17), metar.report_time.hour_of_day);
-        Assert::AreEqual(uint8_t(53), metar.report_time.minute_of_hour);
-    }
+        Assert::AreEqual(uint8_t(8), time.day_of_month);
+        Assert::AreEqual(uint8_t(17), time.hour_of_day);
+        Assert::AreEqual(uint8_t(53), time.minute_of_hour);
+    });
```

Comment:
Should we have something like `Assert::IsTrue(parsed);` like the `METAR_Parser_StationIdentifier` test has?